### DISCUSSION TITLE
Using ImFontGlyphRangesBuilder, I'm failing to add a non Latin character to my Latin character set

### DISCUSSION THREAD

**Discussion by jwalway** [ocornut/imgui#7130] [New Users Build/Link/Run issues ONLY!]
Hello,

I am using ImGui  v1.89.3 on Windows 10, with Visual Studio 2022, OpenGL, and C++. 

I'm attempting to add an extra character to my fonts.  Namely I want a €, euro symbol, along with all the usual Latin characters.    I implemented the code below.  Yet when I run the program the font gives me gibberish for everything.   

Can anyone tell me what I'm doing wrong here?

Thanks!
...John Alway

```
ImFont* _regularFont;

std::string fontName;  
string documentsDir = getDocumentsFolderPath();
fontName = documentsDir + "\\DBus Client\\games\\Verdana.ttf";
 
// Add euro symbol to the glyph range
const char euroSymbol[] = u8"€";
ImVector<ImWchar> ranges;
ImFontGlyphRangesBuilder builder;
builder.AddText(euroSymbol);
builder.AddRanges(ImGui::GetIO().Fonts->GetGlyphRangesDefault()); // Add one of the default ranges, Latin characters
builder.BuildRanges(&ranges);

_regularFont = ImGui::GetIO().Fonts->AddFontFromFileTTF(fontName.c_str(), 16.0f, nullptr, ranges.Data);
ImGui::GetIO().Fonts->Build();
```

**Comment by ocornut** [comment]
A suggestion is:
- First load the default font normally, and load your fonts as 2nd font.
- Then you can more easily use Metrics->Fonts and other [debug tools](https://github.com/ocornut/imgui/wiki/Debug-Tools) to inspect your font data.
- We have a small quirk where in case the atlas gets build again, the range data needs to persist. For now please add "static" keyboard to "ImVector<ImWchar> ranges" to make it persist and ensure this is not your problem for now.

**Reply by jwalway** [reply]
Hello,

Thanks for your response.   I set up the "ranges" so that they are persistent and that fixed the text rendering.  However, the euro symbol did not show up.

I tried the  ImGui::ShowMetricsWindow() function to help me debug.  It's a great tool!  I first tried this with just the Latin characters.  There are 192.   I then added the Euro symbol with builder.AddText(euroSymbol).   ShowMetricsWindow() displayed a second glyphs table, but contained the wrong value.   It filled in the Unicode value 201A, instead of 20AC, the euro symbol Unicode code.   I'm not sure why. The number of glyphs went from 192 to 193.    I also tried adding a second character with the builder.AddChar(0x7262), from the FAQ example, and I didn't see anything for that.   The number of glyphs didn't change.

I attached an image of the ShowMetricsWindow().    See how the second glyphs table shows a ",", i.e. 201A.

![fonts_glyphs_imgui_tool](https://github.com/ocornut/imgui/assets/46982480/e54d5549-c684-4c60-9cf5-49028a1087e4)

**Reply by jwalway** [reply]
I got it working by not using the "u8" symbol in front of the string. i.e. by not doing this builder.AddText(u8"€") and instead doing this builder.AddText("€");

This code worked:
```
ImFontGlyphRangesBuilder builder;
builder.AddText("€");
builder.AddRanges(ImGui::GetIO().Fonts->GetGlyphRangesDefault()); // Add one of the default ranges
builder.BuildRanges(&_rangesRegular);
_regularFont = ImGui::GetIO().Fonts->AddFontFromFileTTF(fontName.c_str(), 16.0f, nullptr, _rangesRegular.Data);
ImGui::GetIO().Fonts->Build();
```

So, it's beautiful that it works.
